includes/alert.js

```diff
@@ -0,0 +1,410 @@
+    // There's nothing fancy about it, and much of it will still work
+    // correctly without JavaScript enabled. However, to offer a slicker
+    // visual user experience, we use BackboneJS to progressively enhance
+    // the static HTML content.
+    //
+    // Backbone helps us structure our interface code into "Views" that
+    // can be composed together. This means one View can contain another
+    // View inside it. Each interface element that we display in the HTML
+    // (like buttons, forms, and so on) are associated with a View at
+    // runtime by code in this file. Interactions that require JavaScript
+    // to work, like showing and hiding dialogues or other visual effects
+    // of a similar nature, are performed by calling methods on the View
+    // objects defined below.
+
+    // --------------------- //
+    // VIEWS (UI components) //
+    // --------------------- //
+
+    // Gotta declare 'em all.
+    var ActivateAlertForm,
+        ChooseTeamsPanel,
+        ContextualAlertButton,
+        ContextualAlertModal,
+        ImmediateAlertButton,
+        SubmittingAlertModal,
+        TimedAlertButton,
+        TimedAlertModal;
+
+    /**
+     * Activate Alert Form
+     *
+     * This is the containing element for the "Activate Alert" screen.
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#accessing-the-activate-alert-screen}
+     */
+    BUOY.Views.ActivateAlertForm = Backbone.View.extend({
+        'events': {
+            'submit': 'handleSubmit'
+        }
+    });
+
+    /**
+     * Immediate Alert Button
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#immediate-alerts}
+     */
+    BUOY.Views.ImmediateAlertButton = Backbone.View.extend({
+        'events': {
+            'click': function () {
+                this.$el.prop('disabled', true); // prevent double-clicks
+            }
+        }
+    });
+
+    /**
+     * Contextual Alert Button
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#contextual-alerts}
+     */
+    BUOY.Views.ContextualAlertButton = Backbone.View.extend({
+        'events': {
+            'click': function () {
+                ChooseTeamsPanel.show();
+                ContextualAlertModal.show();
+            }
+        }
+    });
+
+    /**
+     * Timed Alert Button
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#timed-alerts}
+     */
+    BUOY.Views.TimedAlertButton = Backbone.View.extend({
+        'events': {
+            'click': function () {
+                ChooseTeamsPanel.show();
+                TimedAlertModal.show();
+            }
+        }
+    });
+
+    /**
+     * Contextual Alert Modal
+     *
+     * This is a Bootstrap Modal.
+     *
+     * @see {@link https://getbootstrap.com/javascript/#modals}
+     */
+    BUOY.Views.ContextualAlertModal = Backbone.View.extend({
+        'events': {
+            'click button.btn-success': function (e) {
```

Comment:
I'm slightly bummed about the proliferation of this classname across several Backbone views. My understanding is that the event target is a child element of `this.$el`? I don't actually know what the page markup looks like. Each of these modals has their own `button.btn-success` I assume. I think part of what's grossing me out is the concatenation of the event string and the selector. In theory we could use a variable from `buoy_dom_hooks` to construct the key of the event hash but I think that's overkill. I should just suck it up and accept this convention as the lesser of dom spaghetti evils.